scripts/collate.py

```diff
@@ -1,54 +1,55 @@
-    # Prefix all columns with 'eggNOG.'
-    eggnog = eggnog.rename(lambda x: f'eggNOG.{x}', axis='columns')
-    # Fix some names and remove prefix from ID
-    eggnog = eggnog.rename(columns={"eggNOG.KEGG_ko"    : "eggNOG.KEGG_KO",
-                                    "eggNOG.ID"         : "ID",
-                                    "eggNOG.eggNOG_OGs" : "eggNOG.OGs"})
-    eggnog.set_index("ID", inplace=True)
-    df = eggnog 
+for filename in sys.argv:
+
+    # Make sure file exists
+    if os.path.exists(filename) == False:
+        raise Exception("File {} not found!".format(filename))
+
+    # eggNOG annotations
+    if filename.endswith('_eggNOG.tsv'):
+        eggnog = pd.read_csv(filename, sep="\t", index_col=False)
+        # Prefix all columns with 'eggNOG.'
+        eggnog = eggnog.rename(lambda x: f'eggNOG.{x}', axis='columns')
+        # Fix some names and remove prefix from ID
+        eggnog = eggnog.rename(columns={"eggNOG.KEGG_ko"    : "eggNOG.KEGG_KO",
+                                        "eggNOG.ID"         : "ID",
+                                        "eggNOG.eggNOG_OGs" : "eggNOG.OGs"})
+        eggnog.set_index("ID", inplace=True)
+        df = eggnog
 
-dbcanfile = mydir+"/"+name+"_dbCAN.tsv"
-if os.path.exists(dbcanfile):
-    dbcan = pd.read_csv(dbcanfile, sep="\t", index_col=False)
-    # Prefix eCAMI columns with 'dbCAN.'
-    dbcan = dbcan.rename(columns={"eCAMI.submodule" : "dbCAN.eCAMI_submodule",
-                                  "eCAMI.subfamily" : "dbCAN.eCAMI_subfamily"})
-    dbcan.set_index("ID", inplace=True)
-    df = dbcan 
+    # dbCAN annotations
+    if filename.endswith('_dbCAN.tsv'):
+        dbcan = pd.read_csv(filename, sep="\t", index_col=False)
+        # Prefix eCAMI columns with 'dbCAN.'
+        dbcan = dbcan.rename(columns={"eCAMI.submodule" : "dbCAN.eCAMI_submodule",
+                                      "eCAMI.subfamily" : "dbCAN.eCAMI_subfamily"})
+        dbcan.set_index("ID", inplace=True)
+        df = dbcan
 
-kofamfile = mydir+"/"+name+"_kofam.tsv"
-if os.path.exists(kofamfile):
-    kofam = pd.read_csv(kofamfile, sep="\t", index_col=False)
-    # Prefix all columns with 'kofam.'
-    kofam = kofam.rename(columns={"kofam_KO"      : "kofam.KEGG_KO",
-                                  "kofam_Module"  : "kofam.KEGG_Module",
-                                  "kofam_Pathway" : "kofam.KEGG_Pathway"})
-    kofam.set_index("ID", inplace=True)
-    df = kofam 
+    # kofam annotations
+    if filename.endswith('_kofam.tsv'):
+        kofam = pd.read_csv(filename, sep="\t", index_col=False)
+        # Prefix all columns with 'kofam.'
+        kofam = kofam.rename(columns={"kofam_KO"      : "kofam.KEGG_KO",
+                                      "kofam_Module"  : "kofam.KEGG_Module",
+                                      "kofam_Pathway" : "kofam.KEGG_Pathway"})
+        kofam.set_index("ID", inplace=True)
+        df = kofam
 
```

Comment:
I was about to make a lot of small suggestions, but instead I could pass this then make those changes and also simplify the conditional-djungle in the latter half of the script now that the annotation files are given in the args?

Reply:
Yes, sounds good. When you are done with your new changes, I think we could also rename the script into something like `collate_gene_annotations.py` rather than the non-informative `collate.py`.